app/lib/features/chat_ng/widgets/events/chat_event_item.dart

```diff
@@ -94,20 +96,29 @@ class ChatEventItem extends StatelessWidget {
     String messageId,
     RoomEventItem item,
     bool isUser,
-    bool showAvatar,
+    bool nextMessageGroup,
   ) {
     final msgType = item.msgType();
     final content = item.msgContent();
+    final Map<String, dynamic> metadata = {
+      'isUser': isUser,
+      'nextMessageGroup': nextMessageGroup,
+      'wasEdited': item.wasEdited(),
+      'msgType': msgType,
+    };
 
     // shouldn't happen but in case return empty
     if (msgType == null || content == null) return const SizedBox.shrink();
 
     return switch (msgType) {
-      'm.emote' || 'm.text' => TextMessageEvent(
+      'm.emote' ||
+      'm.notice' ||
+      'm.server_notice' ||
+      'm.text' =>
+        TextMessageEvent(
           roomId: roomId,
           content: content,
-          isUser: isUser,
-          nextMessageGroup: showAvatar,
+          metadata: metadata,
```

Comment:
why not just have them as proper parameters? `Map<String, dynamic>` are hazaradous and easy to get wrong in later version. I'd rather not have `dynamic` passed around if we can avoid it.

Reply:
Yeah I think we can have separate model for that which can contain these information. Because this parameters count will get very large as it grows so I did it as clean way. But it's not type-safe! 👍🏻